src/qml/RelationCombobox.qml

```diff
@@ -22,22 +26,240 @@ Item {
     right: parent.right
     rightMargin: 10
   }
+  height: childrenRect.height + 10
 
   property var currentKeyValue: value
+  property EmbeddedFeatureForm embeddedFeatureForm: embeddedPopup
+
   onCurrentKeyValueChanged: {
     comboBox._cachedCurrentValue = currentKeyValue
     comboBox.currentIndex = featureListModel.findKey(currentKeyValue)
   }
 
-  property EmbeddedFeatureForm embeddedFeatureForm: embeddedPopup
+  EmbeddedFeatureForm{
+      id: addFeaturePopup
 
-  height: childrenRect.height + 10
+      onFeatureSaved: {
+          var referencedValue = addFeaturePopup.attributeFormModel.attribute(relationCombobox._relation.resolveReferencedField(field.name))
+          var index = featureListModel.findKey(referencedValue)
+          if ( index < 0 ) {
+            // model not yet reloaded - keep the value and set it onModelReset
+            comboBox._cachedCurrentValue = referencedValue
+          } else {
+            comboBox.currentIndex = index
+          }
+      }
+  }
+
+
+  Popup {
+    id: searchFeaturePopup
+
+    signal cancel
+    signal apply
+    signal finished
+
+    parent: ApplicationWindow.overlay
+    x: 24
+    y: 24
+    width: parent.width - 48
+    height: parent.height - 48
+    padding: 0
+    modal: true
+    closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutside
+    focus: visible
+
+    onOpened: {
+      searchField.forceActiveFocus()
+    }
+
+    onClosed: {
+      searchField.text = ''
+      searchFeaturePopup.cancel()
+      searchFeaturePopup.finished()
+    }
+
+    Page {
+      anchors.fill: parent
+
+      header: PageHeader {
+        title: fieldLabel
+        showApplyButton: true
+        showCancelButton: true
+        onApply: {
+          searchFeaturePopup.close()
```

Comment:
Will this call cancel/finished through onClosed before apply/finished? Is this intended?

Reply:
Good catch. Thanks. They were triggered twice in a weird order.  I have removed the `finished`, as it will be triggered in the same situations as `closed` .
now we should have:
1)  `apply`|`cancel`
2) `close`